parcels/_datasets/__init__.py

```diff
@@ -0,0 +1,46 @@
+"""
+Datasets compatible with Parcels.
+
+This subpackage uses xarray to generate *idealised* structured and unstructured hydrodynamical datasets that are compatible with Parcels. The goals are three-fold:
+
+1. To provide users with documentation for the types of datasets they can expect Parcels to work with.
+2. To supply our tutorials with hydrodynamical datasets.
+3. To offer developers datasets for use in test cases.
+
+Note that this subpackage is part of the private API for Parcels. Users should not rely directly on the functions defined within this module. Instead, if you want to generate your own datasets, copy the functions from this module into your own code.
+
+Developers, note that only idealised datasets that are (a) quick to generate, and (b) only use dependencies already shipped with Parcels, should be added to this subpackage. Real world datasets should be added to the `parcels-data` repository. No data files should be added to this subpackage.
+
+Parcels Dataset Philosophy
+-------------------------
+
+When adding example datasets, there may be a tension between wanting to add a specific example or wanting to add machinery to generate completely arbitrary datasets (e.g., with different grid resolutions, with different ranges, with different datetimes etc.). There are trade-offs to both approaches:
+
+Working with specific hardcoded examples:
+
+* Pros
+    * the example is stable and self-contained
+    * easy to see exactly what the dataset is, there little to no dependency on other functions defined in the same module
+      * datasets don't "break" due to changes in other functions (e.g., grid edges becoming out of sync with grid centres)
+* Cons
+    * inflexible for use in tests where you want to test a large range of datasets, or you want to test a specific resolution
+
+Working with generated datasets is the opposite of all the above.
+
+Most of the time we only want a single dataset. For example, for use in a tutorial, or for testing a specific feature of Parcels - such as (in the case of structured grids) checking that the grid from a certain vendor is correctly parsed, or checking that indexing is correctly picked up. As such, one should often opt for hardcoded datasets. These are more stable and easier to see exactly what the dataset is. We may have specific examples that become the default "go to" dataset for testing when we don't care about the specific details of the dataset.
+
+Sometimes we may want to test Parcels against a whole range of datasets varying in a certain way - to ensure Parcels works as expected for subspace of possible datasets. For these, we should add machinery to create generated datasets.
+
+Structure
+--------
+
+This subpackage is broken down into structured and unstructured parts. Each of these have common submodules:
+
+* ``providers`` -> hardcoded datasets with the intention of mimicking datasets from a certain provider
```

Comment:
Perhaps another word than `providers`? I'm meaning more NEMO/Mitgcm/CROCO/ROMS/mom5 etc ...

Reply:
`circulationmodels`? See also the comment above about aligning with the text on oceanparcels.org